codex/manifest/manifest.nim

```diff
@@ -25,8 +25,12 @@ import ./coders
 func len*(self: Manifest): int =
   self.blocks.len
 
-func size*(self: Manifest): int =
-  self.blocks.len * self.blockSize
+func originalBytesPadded*(self: Manifest): int =
```

Comment:
Let's add `padded` flag to `size` it should make it more consistent across the other primitives.

Reply:
In old code, Manifest.len is a number of blocks, and Manifest.size is a number of bytes - these names don't look descriptive. So we named the new field originalBytes, and originalBytesPadded name follows the same approach.

I also thought about renaming `len` to blockNum, but since Manifest partially emulates `seq` API, `len` still makes sense.

Now about padding. There are 3 lengths: 
1. original filesize
2. original filesize padded to the block size
3. (available only in protected Manifests) size including ECC data which is always padded

In order to make API somewhat consistent, I proposed that StoreStream can return only file 1 or file 2 of those 3. Size of file 1 is returned by originalBytes, so with originalBytesPadded returning size of file 2, I have everything to implement this concept.

If we sometimes need file 3 (original+parity data), we can implement the following API:
1. StoreStream(pad=false) always returns file 1
2. StoreStream(pad=true) returns file 2 for non-protected Manifest, and file 3 (+ECC) - for protected Manifest

And implement the following helper function:
```Nim
func bytes*(self: Manifest, pad = true): int =
  if self.pad:
    self.len * self.blockSize
  else:
    self.originalBytes
```
that computes how much bytes corresponding StoreStream(Manifest, pad) will return

Reply:
> In old code, Manifest.len is a number of blocks, and Manifest.size is a number of bytes - these names don't look descriptive. So we named the new field originalBytes, and originalBytesPadded name follows the same approach.

`originalBytes` can be considered an internal (private/protected) field for all intents and purposes. To me `len` and `size` make sense because as you noted the manifest is akin to a `seq` in some sense.

> And implement the following helper function:

Yep, looks correct to me and `bytes` makes as much sense as `size` - I'm either way 👍 

> Now about padding. There are 3 lengths:
> 
> 1. original filesize
> 2. original filesize padded to the block size
> 3. (available only in protected Manifests) size including ECC data which is always padded
> 
> In order to make API somewhat consistent, I proposed that StoreStream can return only file 1 or file 2 of those 3. Size of file 1 is returned by originalBytes, so with originalBytesPadded returning size of file 2, I have everything to implement this concept.

I'm proposing that we make no distinction between protected and unprotected manifests and treat them the same:

- unprotected manifests will be either padded or unpadded
- protected manifests will be always padded - i.e. the `pad` flag has no effect
- if only have the protected manifest and you want to read the unprotected data it's always easy to construct an unprotected manifest

The advantage of this is that it reduces the need to handle corner cases and simplifies the problem considerably.

Reply:
ok, so I will change the helper function, that computes how many bytes corresponding StoreStream(Manifest, pad) will return, to:
```Nim
func bytes*(self: Manifest, pad = true): int =
  if self.pad or self.protected:
    self.len * self.blockSize
  else:
    self.originalBytes
```
and it will cover all 3 cases:
1. actual file size for unprotected manifest, pad=false
2. padded file size for unprotected manifest, pad=true
3. size of data+parity blocks for protected manifest, pad ignored

Reply:
I made this change